### PR TITLE
Making range filter generic to any range type.

### PR DESCRIPTION
**Pull request by LucianoPAlmeida**
Hey guys :)) 
Proposal 
I did a PR adding a convenience filter for Range #1605 
But now I think is a better idea make instead of adding a convenience function for any Range type, just make the existing function generic and maintain only on function implementation that can handle any range type for a filter.

### CODE REVIEW

**Comment by sunshinejr** [comment]
@LucianoPAlmeida just wanted to say that this is the first time I'm seeing `RangeExpression` type and I'm amazed. Will probably take a closer look on this and PR later on, but great job on finding it 💯

**Comment by LucianoPAlmeida** [comment]
Hey @sunshinejr I felt the same way when I find out about it, I always amazed at these things on the  Swift :))

File: Sources/Moya/Response.swift [hunk 1]
```diff
@@ -48,24 +48,13 @@ public extension Response {
         - statusCodes: The range of acceptable status codes.
```
Comment (SD10) [inline Sources/Moya/Response.swift]: May want to remove the reference to a closed range in the comment above

**Comment by SD10** [comment]
We'd also need a CHANGELOG entry, this is API breaking cuz we've changed signatures

**Pull request merged**